AtomixTest.shouldFailStartAfterStop flaky

**Summary**

- How often does the test fail? at least once so far top 3 in CW 34
- Does it block your work? no
- Do we suspect that it is a real failure? unknown

**Failures**
Test failed at line 81.

Currently, no further information available.

```
java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Address already in use
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2022)
	at io.atomix.core.AtomixTest.shouldFailStartAfterStop(AtomixTest.java:81)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:157)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Address already in use
```

https://ci.zeebe.camunda.cloud/blue/organizations/jenkins/zeebe-io%2Fzeebe/detail/staging/2084/tests

[test-reports.zip](https://github.com/zeebe-io/zeebe/files/5117158/test-reports.zip)

--- a/test-util/src/main/java/io/zeebe/test/util/TestEnvironment.java
+++ b/test-util/src/main/java/io/zeebe/test/util/TestEnvironment.java
@@ -16,6 +16,11 @@ public class TestEnvironment {
   private static final String TEST_FORK_NUMBER_PROPERTY_NAME = "testForkNumber";
   private static final String TEST_MAVEN_ID_PROPERTY_NAME = "testMavenId";
 
+  /**
+   * Returns the test fork number
+   *
+   * @return test fork number
+   */
   public static int getTestForkNumber() {
     int testForkNumber = 0;
     try {
@@ -34,6 +39,12 @@ public class TestEnvironment {
     return testForkNumber;
   }
 
+  /**
+   * Returns the test maven ID that maps to a test stage in Jenkins (e.g. junit = 1; it = 2; junit8
+   * = 3
+   *
+   * @return test maven ID that maps to a test stage in Jenkins (e.g. junit = 1; it = 2; junit8 = 3
+   */
   public static int getTestMavenId() {
     int testMavenId = 0;
     try {

--- a/test-util/src/main/java/io/zeebe/test/util/socket/SocketUtil.java
+++ b/test-util/src/main/java/io/zeebe/test/util/socket/SocketUtil.java
@@ -17,13 +17,18 @@ public final class SocketUtil {
 
   private static final String DEFAULT_HOST = "localhost";
   private static final int BASE_PORT = 25600;
-  private static final int RANGE_SIZE = 100;
+  private static final int PORT_RANGE_PER_TEST_FORK = 100;
+  private static final int MAX_TEST_FORKS_PER_STAGE = 39;
+  private static final int PORT_RANGE_PER_TEST_STAGE =
+      PORT_RANGE_PER_TEST_FORK * MAX_TEST_FORKS_PER_STAGE; // 3900
+  private static final int MAX_TEST_STAGES = 10;
 
   private static final int TEST_FORK_NUMBER;
   private static final PortRange PORT_RANGE;
 
   static {
     final int testForkNumber = TestEnvironment.getTestForkNumber();
+    // test stage in Jenkins (junit8, junit, it)
     final int testMavenId = TestEnvironment.getTestMavenId();
 
     LOG.info(
@@ -32,13 +37,16 @@ public final class SocketUtil {
         testMavenId);
 
     // ensure limits to stay in available port range
-    assert testForkNumber < 39 : "System property test fork number has to be smaller then 39";
-    assert testMavenId < 10 : "System property test maven id has to be smaller then 10";
+    assert testForkNumber < MAX_TEST_FORKS_PER_STAGE
+        : "System property test fork number has to be smaller than " + MAX_TEST_FORKS_PER_STAGE;
+    assert testMavenId < MAX_TEST_STAGES
+        : "System property test maven id has to be smaller than " + MAX_TEST_STAGES;
 
-    final int testOffset = testForkNumber * 10 + testMavenId;
+    final int testOffset =
+        testMavenId * PORT_RANGE_PER_TEST_STAGE + testForkNumber * PORT_RANGE_PER_TEST_FORK;
+    final int min = BASE_PORT + testOffset;
+    final int max = min + PORT_RANGE_PER_TEST_FORK;
 
-    final int min = BASE_PORT + testOffset * RANGE_SIZE;
-    final int max = min + RANGE_SIZE;
     TEST_FORK_NUMBER = testForkNumber;
     PORT_RANGE = new PortRange(DEFAULT_HOST, TEST_FORK_NUMBER, min, max);
   }

